dex-template/__DexName__.ts.template

```diff
@@ -0,0 +1,224 @@
+    this.logDecoder = (log: Log) => this.interface.parseLog(log);
+    this.addressesSubscribed = [/* subscribed addresses */];
+
+    // Add handlers
+    this.handlers['myEvent'] = this.handleMyEvent.bind(this);
+  }
+
+  /**
+   * The function is called everytime any of the subscribed 
+   * addresses release log. The function accepts the current
+   * state, updates the state according to the log, and returns
+   * the updated state. 
+   * @param state - Current state of event subscriber
+   * @param log - Log released by one of the subscribed addresses
+   * @returns Updates state of the event subscriber after the log
+   */
+  protected processLog(
+    state: DeepReadonly<PoolState>,
+    log: Readonly<Log>,
+  ): DeepReadonly<PoolState> | null {
+    try {
+      const event = this.logDecoder(log);
+      if (event.name in this.handlers) {
+        return this.handlers[event.name](
+          event,
+          state,
+          log,
+        );
+      }
+      return state;
+    } catch (e) {
+      this.logger.error(
+        `Error_${this.parentName}_processLog could not parse the log with topic ${log.topics}:`,
+        e,
+      );
+      return null;
+    }
+  }
+
+  /**
+   * The function generates state using on-chain calls. This
+   * function is called to regenrate state if the event based 
+   * system fails to fetch events and the local state is no 
+   * more correct. 
+   * @param blockNumber - Blocknumber for which the state should
+   * should be generated
+   * @returns state of the event subsriber at blocknumber
+   */
+  async generateState(blockNumber: number): Promise<Readonly<PoolState>> {
+    // TODO: complete me!
+  }
+}
+
+export class __DexName__
+  extends SimpleExchange
+  implements IDex<__DexName__Data>
+{
+  protected eventPools: __DexName__EventPool;
+
+  readonly hasConstantPriceLargeAmounts = false;
+
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(__DexName__Config);
+
+  logger: Logger;
+
+  constructor(
+    protected network: Network,
+    protected dexKey: string,
+    protected dexHelper: IDexHelper,
+    // TODO: add any additional optional params to support other fork DEXes
+  ) {
+    super(dexHelper.augustusAddress, dexHelper.provider);
+    this.logger = dexHelper.getLogger(dexKey);
+    this.eventPools = new __DexName__EventPool(
+      dexKey,
+      network,
+      dexHelper,
+      this.logger,
+    );
+  }
+
+  // Initialize pricing is called once in the start of 
+  // pricing service. It is intended to setup the integration
+  // for pricing requests. It is optional for a DEX to 
+  // implement this function
+  async initializePricing(blockNumber: number) {
+    // TODO: complete me!
+  }
+
+  // Returns the list of contract adapters (name and index)
+  // for a buy/sell. Return null if there are no adapters.
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    // TODO: complete me!
```

Comment:
I think this could have an implementation in the template.

Reply:
fixed